bids/analysis/transformations/base.py

```diff
@@ -43,9 +43,14 @@ class Transformation(metaclass=ABCMeta):
 
     # A tuple indicating which arguments give the names of variables that must
     # all be aligned with one another (i.e., onsets and durations match
-    # perfectly) before processing. Defaults to None.
+    # perfectly) before processing.
+    # If True, will require all variables to be aligned. Defaults to None.
     _align = None
 
+    # A boolean indicating if auto-alignment by densification should occur
+    # if alignment is required, but inputs are not aligned.
+    _force_dense_align = False
```

Comment:
Do we need a separate variable for this? if we could make `_align` take values `None`, `True`, or 'dense'`, that might be cleaner. Haven't thought through whether that would work though.

Reply:
I think we do because `_align` can be boolean, or a tuple which indicates which variable indicates which need to be aligned.

Reply:
For example in `orthogonalize`, it takes the value `other` so that the columns specified in that variable are the ones that need to be aligned (not all columns, which may not be required for this transformation).

Reply:
Ah, right. In that case, for clarity we we should maybe call the variables `_alignment_mode` and `aligned_variables`, where the former takes the 3 values I suggested, and the latter defaults to `None` but can be set to a tuple of variable names.

Reply:
I think what makes more sense to me is : `_align_required` : False, True, and `force`, such that if its True it will throw an error, instead of forcing, correct?

`_aligned_variables` 'all', None, or a tuple with names.

Reply:
Okay, clarification: the original behavior allowed for either: None, a single string which indicated which column to find names of columns to be aligned (e.g. 'other'), or a tuple the explicit names of columns to be aligned.

and this looks like the logic that does that:
```
        align_variables = [listify(self.kwargs[v])
                           for v in listify(self._aligned_required)
                           if v in self.kwargs]
        align_variables = list(itertools.chain(*align_variables))
        align_variables = [self.collection[c] for c in align_variables if c]
```

I'll need to work through that and separate the logic into two variables.

Reply:
Sure that works.

Reply:
okay done! tests passing locally